File: codesignal/cs-strings-construction/strings-construction.js
/* eslint-disable no-unused-vars */

function stringsConstruction(a, b) {
  let counter = 0;
  const bStrList = b.split('');

  while ((bStrList.length - a.length) >= 0) {
    for (let i = 0; i < a.length; i++) {
      const indexMatch = bStrList.indexOf(a[i]);
      if (indexMatch >= 0) {
        bStrList.splice(indexMatch, 1);
      } else {
        return counter;
      }
    }
    counter++;
  }
  return counter;
}

File: codesignal/cs-strings-construction/strings-construction.py
def stringsConstruction(a, b):
    counter = 0
    bStrList = list(b)

    while len(bStrList) - len(a) >= 0:
        for i in a:
            if i in bStrList:
                bStrList.remove(i)
            else:
                return counter

        counter += 1

    return counter
